porting/halium_7-1/Configuring.rst

```diff
@@ -118,6 +118,218 @@ At the end of the file, append this::
+
+    python2 ./gentree.py --copy-list ./copy-list --integrate --clean --git-revision v4.2 ~/kernel-backports/linux-next ~/halium/kernel/[VENDOR]/[MODEL_NAME]
+
+You will get error messages at the end and if they do not mention the backport Makefile and/or Kconfig, this means that the script has aborted before completion. You will then need to determine the cause and retry. 
+
+If the final error message concerns the Makefile and includes info about having generated a file named ``Makefile.rej``, this means you will find information in this file about changes that did not complete successfully, but which you can apply yourself. These need to be completed before proceeding with the build.
+
+Apply security patch
+^^^^^^^^^^^^^^^^^^^^
+
+An additional `generic security patch <https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/patch/?id=8a7b081660857a80c3efc463b3da790c4fa0c801>`_ needs to be applied. 
+
+Edit kernel defconfig
+^^^^^^^^^^^^^^^^^^^^^
+
+Your kernel config file (defconfig) needs to be modified in order for the backported driver and protocol code to be activated.
+
+Start by locating all lines beginning with ``CONFIG_BT_`` and move these to the end of the file. Collecting them there makes the subsequent steps somewhat easier by helping to keep track of the changes you make.
+
+Next, deactivate all that are activated, *i.e.* do not have a leading ``#``, by inserting this leading ``#``. At the same time, for each one, add a corresponding one beginning with ``CONFIG_BACKPORT_BT_``, *e.g.*::
+
+    CONFIG_BT=y
+
+becomes::
+
+    #CONFIG_BT=y
+
+and then insert the corresponding line for backports::
+
+    CONFIG_BACKPORT_BT=y
+
+Some more configuration settings are necessary, depending on your device. You will likely need the settings listed here, but additional ones can also be necessary::
+
+    #Depending options for new stuff from backports
+    #CONFIG_CRC16=y
+    CONFIG_CRYPTO=y
+    CONFIG_CRYPTO_BLKCIPHER=y
+    CONFIG_CRYPTO_AES=y
+    CONFIG_CRYPTO_CMAC=y
+    CONFIG_CRYPTO_HMAC=y
+    CONFIG_CRYPTO_ECB=y
+    CONFIG_CRYPTO_SHA256=y
+    CONFIG_CRYPTO_USER_API=y
+    CONFIG_CRYPTO_USER_API_HASH=y
+    CONFIG_CRYPTO_USER_API_SKCIPHER=y
+    #CONFIG_TTY=y
+
+At this point, check for any remaining settings you :ref:`recorded from your original defconfig <BT-driver>`, making sure not to forget your device's bluetooth driver. 
+
+Any remaining settings which are dependent upon CONFIG_BT will no longer have any effect. They need to be replaced. If you find a setting that is not yet included after the changes you made (above) this probably means it and the corresponding source file(s) will have to be migrated from their original location to the corresponding location under ``backport/bluetooth/``. The files ``Makefile`` and ``Kconfig`` need to be edited to include this missing setting or else they will not be built. Check the corresponding files in the original location for the necessary settings.
```

Comment:
'If you find' ... how would I find it?

Can we remove the '(above)'?

Reply:
OK. Let me rethink this a bit. The explanation is there, but oviously not clear enough.

Reply:
I have now reformulated a bit.

Reply:
yes, much clearer. thanks!